enumerable.rb

```diff
@@ -0,0 +1,124 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    my_array = is_a?(Range) ? to_a : self
+
+    my_array.size.times { |n| yield (my_array[n]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    my_array = is_a?(Range) ? to_a : self
+
+    my_array.size.times { |n| yield my_array[n], n }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result_array = []
+    my_each { |n| result_array << n if yield(n) }
+    result_array
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all(arg = nil)
+    result = true
+
+    my_each do |i|
+      result = if arg
+                 arg.is_a?(Class) ? i.is_a?(arg) : i.to_s.match?(arg.to_s)
+               elsif block_given?
+                 yield i
+               else
+                 i == false || i.nil? ? false : true
+               end
+      return false unless result
+    end
+
+    result
+  end
+
+  def my_any(arg = nil)
+    result = false
+    my_each do |i|
+      result = if arg
+                 arg.is_a?(Class) ? i.is_a?(arg) : i.to_s.match?(arg.to_s)
+               elsif block_given?
+                 yield i
+               else
+                 i == false || i.nil? ? false : true
+               end
+      return true if result
+    end
+    result
+  end
+
+  def my_none(arg = nil)
+    result = true
+    my_each do |i|
+      result = if arg
+                 arg.is_a?(Class) ? i.is_a?(arg) : i.to_s.match?(arg.to_s)
+               elsif block_given?
+                 !yield i
+               else
+                 !i == false || i.nil? ? false : true
+               end
+      return true if result
+    end
+    result
+  end
```

Comment:
### #my_none? 
* [ ] All Boolean function should end with a question mark for best practices
* [ ] Returns false if the block ever returns true for all elements in a range
* [ ] When no block or argument is given returns false only if one of the collection members is true
* [ ] When a Regex is passed as an argument returns true if none of the collection matches the Regex
* [ ] When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
* [ ] When a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

Reply:
fixed the logic
![passed2](https://user-images.githubusercontent.com/51087315/83947999-2c5f8280-a823-11ea-8f5d-0b0d5df3da1d.png)

Reply:
all rub doc tests passed